runner.py

```diff
@@ -0,0 +1,285 @@
+"""
+Runner is a wrapper for any command with some useful options.
+
+Usage: runner.py [OPTIONS] '[CMD]'
+"""
+
+import click
+import subprocess
+import psutil
+import time
+import logging
+
+
+class Command(object):
+    """A command wrapper to run the command."""
+
+    def __init__(
+        self,
+        cmd=[],
+        attempts=1,
+        sys_trace=False,
+        call_trace=False,
+        log_trace=False
+    ):
+        """
+        Initiate the command wrapper.
+
+        Parameters
+        ----------
+        cmd : list
+            The command to run (default is empty).
+        attempts : int
+            Number of allowed failed attempts (default is 1).
+        sys_trace : bool
+            A flag indicating if system trace should be enabled (default is False).
+        call_trace : bool
+            A flag indicating if call trace should be enabled (default is False).
+        log_trace : bool
+            A flag indicating if log trace should be enabled (default is False).
+        """
+        if len(cmd) == 0:
+            raise ValueError('No command specified')
+
+        if call_trace:
+            cmd = ['strace'] + cmd
+
+        self.__cmd = cmd
+        self.__attempts = attempts
+        self.__call_trace = call_trace
+        self.__sys_trace = sys_trace
+        self.__log_trace = log_trace
+        self.__return_codes = {}
+        self.__captured_sys_trace = None
+
+    def execute(self, repeat_times=1):
+        """Executing a command and waiting for it to complete.
+
+        Parameters
+        ----------
+        repeat_times : int
+            Number of times to run command (default is 1).
+        """
+        logging.debug('execute()')
+
+        for _ in range(repeat_times):
```

Comment:
1.use context managers where ever possible unless you have a good reason not to https://book.pythontips.com/en/latest/context_managers.html
2. i would seperate the retrying mechanisim from the actual execuation. 
3. `__executed` should be either a callback or a seperate concern. what happens after the command invokation is not the concern of the command